### ISSUE TITLE
UserWarning: Copy construct from tensor - samplers.py:321

### ISSUE THREAD

**Issue by Chryseus** [comfyanonymous/ComfyUI#1842]
`C:\ComfyUI\comfy\samplers.py:321: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).`
Warning appears once on first generation as of commit 7fbb217 using xformers, no other problems noticed, using PyTorch 2.1.0 cu121, xformers 0.0.22.post7.

**Comment by NeedsMoar** [comment]
I haven't seen this on a custom-built xformers (from when flash attention needed custom builds on windows) 
xFormers 0.0.23+d20354a.d20231007 so it may be something on their end.  The actual reason for that message is slightly incredibly stupid:

`When working with tensors prefer using torch.Tensor.clone(), torch.Tensor.detach(), and torch.Tensor.requires_grad_() for readability. Letting t be a tensor, torch.tensor(t) is equivalent to t.clone().detach(), and torch.tensor(t, requires_grad=True) is equivalent to t.clone().detach().requires_grad_(True).`

Which is "more readable" since "torch.tensor(t)" and "Tensor.detach()" both remove the instance from the execution graph and torch.tensor(t) defines __deepcopy__ so it actually makes a clone of the data rather than aliasing it in place like the usual python class.  In other words they want you writing out both the stupid side effect that should either be clear from the act of copying or not happen until post-construction by sane programming practices,  and the thing that should be happening when you construct an object from another instance of that type of object anyway but doesn't unless that type defines __deepcopy__ which is completely arbitrary.   Define the named parameter with a method call instead, too, because we've already gone this far into silly.  

Anyway they may have added this warning in that release of xformers then removed it since it's documented in the torch manual anyway or it may be in the version of torch you're using, I'm not entirely sure.  I'm guessing xformers since my torch version is just 2.1.0+cu121 which is the same as you're using, and post7 has flash_attention2 built in again.  In any case I don't think comfy touches the graph stuff in torch anyway and I'm almost certain it disables autograd for the samplers so the default torch.tensor(foo) call does what you'd expect, and in any case it's just asking for a more verbose version of that line of code, not saying there's anything wrong with it per se.
